### PR TITLE
Fixed not removed deleted original files

### LINKED ISSUE #1679: Delete old files from media collections
When files are deleted in the backend, they are not removed from the default `uploads/media' folder.

Did I miss some configuration/option to turn this on? Or is this intended to work like this?

PS: Guys, I wonder why never came across your great work of software. You thought about so many great things and the interface is gorgeous – especially in comparison with other stuff out there. Thank you for making such great software free to use for everyone!

### PR DESCRIPTION
**Pull request by wachterjohannes**
__tasks:__

- [x] test coverage
- [ ] gather feedback for my changes

__informations:__

| q                | a
| ---------------- | ---
| Fixed tickets    | fixes #1679 
| BC breaks        | none 
| Documentation PR | none

### CODE REVIEW

**Comment by wachterjohannes** [comment]
@danrot this was an easy fix! should we also take that in 1.1?

File: src/Sulu/Bundle/MediaBundle/Media/Storage/LocalStorage.php [hunk 1]
```diff
@@ -34,20 +36,22 @@ class LocalStorage implements StorageInterface
...
      */
-    public function __construct($uploadPath, $segments, $logger = null)
+    protected $logger;
+
+    public function __construct($uploadPath, $segments, Filesystem $filesystem, $logger = null)
     {
         $this->uploadPath = $uploadPath;
         $this->segments = $segments;
         $this->logger = $logger ?: new NullLogger();
+
```
Comment (danrot) [inline src/Sulu/Bundle/MediaBundle/Media/Storage/LocalStorage.php]: Why is there a new line here?

Comment (wachterjohannes) [inline src/Sulu/Bundle/MediaBundle/Media/Storage/LocalStorage.php]: when i should be honest (: i have no idea!

File: src/Sulu/Bundle/MediaBundle/Tests/Functional/Controller/MediaControllerTest.php [hunk 2]
```diff
@@ -929,6 +929,8 @@ public function testDeleteById()
         $response = json_decode($client->getResponse()->getContent());
         $this->assertEquals(5015, $response->code);
         $this->assertTrue(isset($response->message));
+
+        $this->assertFalse(file_exists( __DIR__ . '/../../uploads/media/1/photo.jpeg'));
```
Comment (danrot) [inline src/Sulu/Bundle/MediaBundle/Tests/Functional/Controller/MediaControllerTest.php]: You could use `assertFileExists` here instead.

**Comment by danrot** [comment]
Yeah, we can take it into 1.1.

**Comment by wachterjohannes** [comment]
@danrot fixed

**Comment by danrot** [comment]
This PR only removes the converted files in `web/uploads/media`, but not the ones in `uploads/media`, where the originals are stored.

File: src/Sulu/Bundle/MediaBundle/Media/Manager/MediaManager.php [hunk 3]
```diff
@@ -813,6 +813,8 @@ public function delete($id, $checkSecurity = false)
                     $fileVersion->getName(),
                     $fileVersion->getStorageOptions()
                 );
+
+                $this->storage->remove($fileVersion->getStorageOptions());
```
Comment (wachterjohannes) [inline src/Sulu/Bundle/MediaBundle/Media/Manager/MediaManager.php]: @danrot this also removes the originals ...

Comment (danrot) [inline src/Sulu/Bundle/MediaBundle/Media/Manager/MediaManager.php]: Hm, strange, now it works... ok, everything fine then.

**Pull request merged**